rundeckapp/grails-app/controllers/rundeck/controllers/FrameworkController.groovy

```diff
@@ -928,8 +930,12 @@ class FrameworkController extends ControllerBase implements ApplicationContextAw
         } else if (params.description && !(params.description =~ FrameworkResource.VALID_RESOURCE_DESCRIPTION_REGEX)) {
             projectDescriptionError = message(code: "project.description.can.only.contain.these.characters")
             errors << projectDescriptionError
-        } else if (framework.getFrameworkProjectMgr().existsFrameworkProject(project)) {
-            projectNameError = "Project already exists: ${project}"
+        } else if (projectManager.isFrameworkProjectDisabled(project)) {
+            projectNameError = message(code: "project.disabled", args: [project])
+            log.error(projectNameError)
+            errors << projectNameError
+        } else if (projectManager.existsFrameworkProject(project)) {
+            projectNameError = message(code: "project.exists", args: [project])
```

Comment:
wow, I think this validation checks (name, exist, disabled) should go much earlier in this method, so we can fail fast without doing all of the other stuff.  Not directly related to your change, but seems like basic validation that should come early